### PR TITLE
feat: maker UI integration

### PR DESCRIPTION
**Pull request by offerijns**
- Add Maker overview bar.
- Fix values shown across pool pages, incl. oversubscribed label.
- Add ability to raise/sink creditline.
- Add maintenance mode.
- Update tinlake bot interval to fix duplication issue.

### CODE REVIEW

File: tinlake-bot/src/index.ts [hunk 1]
```diff
@@ -41,14 +41,14 @@ const run = async () => {
   })
   cronJobs.set('closePools', closePoolsTask)
 
-  let submitSolutionsTask = new CronJob(CronExpression.EVERY_10_MINUTES, async () => {
-    // Submit solutions every 10 minutes
+  let submitSolutionsTask = new CronJob(CronExpression.EVERY_30_MINUTES, async () => {
+    // Submit solutions every 15 minutes
```
Comment (philipstanislaus) [inline tinlake-bot/src/index.ts:45]:
```suggestion
    // Submit solutions every x minutes
```

File: tinlake-bot/src/index.ts [hunk 2]
```diff
@@ -41,14 +41,14 @@ const run = async () => {
...
     await submitSolutions(pools, provider, signerWithProvider)
   })
   cronJobs.set('submitSolutions', submitSolutionsTask)
 
-  let executePoolsTask = new CronJob(CronExpression.EVERY_5_MINUTES, async () => {
-    // Execute pool epochs every 5 minutes
+  let executePoolsTask = new CronJob(CronExpression.EVERY_30_MINUTES, async () => {
+    // Execute pool epochs every 15 minutes
```
Comment (philipstanislaus) [inline tinlake-bot/src/index.ts:51]:
```suggestion
    // Execute pool epochs every x minutes
```

File: tinlake-ui/components/Overview/OverviewHeader.tsx [hunk 3]
```diff
@@ -68,61 +77,104 @@ const OverviewHeader: React.FC<Props> = (props: Props) => {
...
+            <h2>
+              {addThousandsSeparators(toPrecision(baseToDisplay(poolData?.maker?.remainingCredit || new BN(0), 18), 0))}{' '}
+              <MakerUnit>DAI</MakerUnit>{' '}
+            </h2>
+          </MakerMetric> */}
+          <MakerMetric margin={{ left: 'auto' }} style={{ borderRight: '1px solid #fff' }}>
+            <h3>Current Debt</h3>
+            <h2>
+              {addThousandsSeparators(toPrecision(baseToDisplay(poolData?.maker?.art || new BN(0), 18), 0))}{' '}
+              <MakerUnit>DAI</MakerUnit>{' '}
```
Comment (philipstanislaus) [inline tinlake-ui/components/Overview/OverviewHeader.tsx:159]: Shall we use pool currencies here as we do in other places?

File: tinlake-ui/containers/Investment/View/EpochOverview.tsx [hunk 4]
```diff
@@ -258,7 +256,7 @@ const EpochOverview: React.FC<Props> = (props: Props) => {
               </TableBody>
             </Table>
 
-            {isAdmin && epochData && (
+            {epochData && (
```
Comment (philipstanislaus) [inline tinlake-ui/containers/Investment/View/EpochOverview.tsx:259]: Is this change intended? I. e. everyone should be able to close the epoch now?

File: tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx [hunk 5]
```diff
@@ -25,13 +27,47 @@ const MaxReserveForm: React.FC<Props> = (props: Props) => {
...
+
+    if (creditlineValue && mat && creditlineValue !== (mat || new BN(0)).toString()) {
+      const currentCreditline = poolData?.maker?.creditline?.toString()
+      const amount = new BN(creditlineValue).gt(new BN(currentCreditline))
+        ? new BN(creditlineValue).sub(new BN(currentCreditline))
+        : new BN(currentCreditline).sub(new BN(creditlineValue))
+      const valueToDecimal = new Decimal(baseToDisplay(creditlineValue, 18)).toDecimalPlaces(4)
+      const formatted = addThousandsSeparators(valueToDecimal.toString())
+
+      console.log(`${currentCreditline} => ${creditlineValue}`)
```
Comment (philipstanislaus) [inline tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx:55]: Remove this?

File: tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx [hunk 6]
```diff
@@ -41,26 +77,75 @@ const MaxReserveForm: React.FC<Props> = (props: Props) => {
...
+    .div(poolData?.minJuniorRatio || new BN(0))
+    .mul(new BN(10).pow(new BN(27)).sub(poolData?.minJuniorRatio || new BN(0)))
+    .div(new BN(10).pow(new BN(27)))
+  const maxCreditline =
+    mat && !mat.isZero()
+      ? maxDropDAI
+          .sub(effectiveDropBalanceDAI)
+          .mul(new BN(10).pow(new BN(27)))
+          .div(mat)
+      : new BN('0')
```
Comment (philipstanislaus) [inline tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx:99]: For the sanity of the reader, we might want to create some helpers for those functions 😅

File: tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx [hunk 7]
```diff
@@ -41,26 +77,75 @@ const MaxReserveForm: React.FC<Props> = (props: Props) => {
...
           disabled={status === 'pending' || status === 'unconfirmed'}
         />
-        <Button primary label="Set" onClick={save} disabled={status === 'pending' || status === 'unconfirmed'} />
+        <Button
+          primary
+          label="Set"
+          onClick={save}
+          disabled={
+            status === 'pending' ||
+            status === 'unconfirmed' ||
```
Comment (philipstanislaus) [inline tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx:145]: Here (and also with the other status fields above), we should probably consider credit line tx's status as well, right?

**Review by philipstanislaus — APPROVED** [review]
Nice, and good idea to have the mainentance mode flags available. Could you add an example how to set them to the README though? Thanks!

💪

File: tinlake-ui/components/Overview/OverviewHeader.tsx (hunk 3 above)
Comment (offerijns) [reply tinlake-ui/components/Overview/OverviewHeader.tsx:159]: The Maker Debt Ceiling will always be DAI, so I don't think that makes sense in this case.

File: tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx (hunk 6 above)
Comment (offerijns) [reply tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx:99]: Yeah absolutely.. This is actually not properly working yet, once I fix it, I'll also see if I can refactor it.

File: tinlake-ui/containers/Investment/View/EpochOverview.tsx (hunk 4 above)
Comment (offerijns) [reply tinlake-ui/containers/Investment/View/EpochOverview.tsx:259]: Yeah that's intended :) Since it's working properly now, everyone can do it.

File: tinlake-ui/components/Overview/OverviewHeader.tsx (hunk 3 above)
Comment (philipstanislaus) [reply tinlake-ui/components/Overview/OverviewHeader.tsx:159]: Makes sense 👍

File: tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx (hunk 7 above)
Comment (offerijns) [reply tinlake-ui/containers/Loan/Overview/MaxReserveForm.tsx:145]: Indeed!
